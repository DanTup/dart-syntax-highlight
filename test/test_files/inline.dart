// Copyright (c) 2023, the Dart project authors. Please see the AUTHORS file
// for details. All rights reserved. Use of this source code is governed by a
// BSD-style license that can be found in the LICENSE file.

inline class MyInt {
  final int it;
  MyInt(this.it);
  void foo() { print('MyInt.foo'); }
}

final inline = 1;

int get inline => 1;
set inline(int value) {}

void inline() {
  var inline = 1;
}
