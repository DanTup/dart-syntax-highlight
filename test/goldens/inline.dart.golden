>// Copyright (c) 2023, the Dart project authors. Please see the AUTHORS file
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ comment.line.double-slash.dart
>// for details. All rights reserved. Use of this source code is governed by a
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ comment.line.double-slash.dart
>// BSD-style license that can be found in the LICENSE file.
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ comment.line.double-slash.dart
>
>inline class MyInt {
#^^^^^^^^^^^^ keyword.declaration.dart
#             ^^^^^ support.class.dart
>  final int it;
#  ^^^^^ storage.modifier.dart
#        ^^^ support.class.dart
#              ^ punctuation.terminator.dart
>  MyInt(this.it);
#  ^^^^^ support.class.dart
#        ^^^^ variable.language.dart
#            ^ punctuation.dot.dart
#                ^ punctuation.terminator.dart
>  void foo() { print('MyInt.foo'); }
#  ^^^^ storage.type.primitive.dart
#       ^^^ entity.name.function.dart
#               ^^^^^ entity.name.function.dart
#                     ^^^^^^^^^^^ string.interpolated.single.dart
#                                 ^ punctuation.terminator.dart
>}
>
>final inline = 1;
#^^^^^ storage.modifier.dart
#             ^ keyword.operator.assignment.dart
#               ^ constant.numeric.dart
#                ^ punctuation.terminator.dart
>
>int get inline => 1;
#^^^ support.class.dart
#    ^^^ keyword.declaration.dart
#               ^^ keyword.operator.closure.dart
#                  ^ constant.numeric.dart
#                   ^ punctuation.terminator.dart
>set inline(int value) {}
#^^^ keyword.declaration.dart
#    ^^^^^^ entity.name.function.dart
#           ^^^ support.class.dart
>
>void inline() {
#^^^^ storage.type.primitive.dart
#     ^^^^^^ entity.name.function.dart
>  var inline = 1;
#  ^^^ storage.type.primitive.dart
#             ^ keyword.operator.assignment.dart
#               ^ constant.numeric.dart
#                ^ punctuation.terminator.dart
>}
